### ISSUE TITLE
Custom specific markers/labels in Y axis dynamic based on event.

### ISSUE THREAD

**Issue by dhanjit** [oxyplot/oxyplot#1375]
### Feature description

Marking specific points in the y axes.
basically I want to color/format or highlight specific y axis value  in the plot depending on events (mouse over in my case). I already how to bind mouse hover and process it:

```c#
PlotController.Bind(new OxyMouseEnterGesture(), new DelegatePlotCommand<OxyMouseEventArgs>(
    new Action<IPlotView, IController, OxyMouseEventArgs>((view, controller, args) => {
          PlotCommands.HoverSnapTrack.Execute(view, controller, args);
          controller.AddHoverManipulator(view, new CustomMouseManipulatorDerived(view), args);
    })
));
```
And then do things in `CustomMouseManipulatorDerived.Delta()` function.

What I do not know is how do I mark specific points in the Y axis?

![image](https://user-images.githubusercontent.com/2098349/67013510-d95c8200-f110-11e9-8c4e-cb3afa445632.png)

Here, the y axis has red markers which would be some default markings already exisisting.
The green and the purple markings/labels would be dynamic and change based on events. How do i achieve this?

**Comment by objorke** [comment]
I think you need to create a custom axis. Derive a new class from any of the axis classes.
For examples how to do this, see the `CustomAxisExamples` in the ExampleLibrary.

**Comment by dhanjit** [comment]
@objorke Hi thanks. I was able to do that. However there is a unique issue. When exactly does Axis.Render get called? Can I call it when i want my y axis to be refreshed so that the points get drawn (when i move the tracker/cursor)? If so, the rendercontext needs to be given which is i believe private in model (I checked the source... that's what it seems). there is no way to get Oxyplot.Wpf.CanvasRenderContext which is the type i get when my CustomYAxis.Render  (inherting from LinearAxis) is called.

The chart is on grid column 3 and stackpanel on grid column 1. when I don't fix width of the sidepanel, everything related to plot seems to get re-rendered hence the axis too get re-rendered and hence my drawn symbols/points show up on every hover/move of cursor/tracker. 
But when I fix width of the stackpanel on grid column 1, the plot doesn't seem to get re-rendered... Now the calls to CustomYAxis.Render only happen when I scroll or zoom etc... This is quite confusing honestly. Quite sure it's not a issue with oxyplot but if I can know when and how to force refresh of yaxis it would solve my issue.
